jupyter_server/base/zmqhandlers.py

```diff
@@ -254,6 +259,31 @@ def _on_zmq_reply(self, stream, msg_list):
         else:
             self.write_message(msg, binary=isinstance(msg, bytes))
 
+    def _on_zmq_reply_0_0_1(self, stream, msg_list):
+        # Sometimes this gets triggered when the on_close method is scheduled in the
```

Comment:
Can we extract this check to a helper function so it can be shared?

Reply:
I like the idea of having the legacy protocol untouched as much as possible, and just *adding* code for the new protocol. But I understand if you prefer not to repeat anything.